src/ApparentHorizons/YlmSpherepack.cpp

```diff
@@ -517,44 +518,65 @@ YlmSpherepack::first_and_second_derivative(
   return result;
 }
 
-YlmSpherepack::InterpolationInfoPerPoint::InterpolationInfoPerPoint(
+template <typename T>
+YlmSpherepack::InterpolationInfo<T>::InterpolationInfo(
     const size_t m_max, const std::vector<double>& pmm,
-    const std::array<double, 2>& target)
-    : cos_theta(cos(target[0])),
-      cos_m_phi(m_max + 1),
-      sin_m_phi(m_max + 1),
-      pbar_factor(m_max + 1) {
-  const double sin_theta = sin(target[0]);
-  const double phi = target[1];
+    const std::array<T, 2>& target_points)
+    : cos_theta(cos(target_points[0])),
+      cos_m_phi(DynamicBuffer<T>(m_max + 1, get_size(target_points[0]))),
+      sin_m_phi(DynamicBuffer<T>(m_max + 1, get_size(target_points[0]))),
+      pbar_factor(DynamicBuffer<T>(m_max + 1, get_size(target_points[0]))),
+      num_points_(get_size(target_points[0])) {
+  const auto& theta = target_points[0];
+  const auto& phi = target_points[1];
+
+  ASSERT(get_size(theta) == get_size(phi),
+         "Size mismatch for thetas and phis: " << get_size(theta) << " and "
+                                               << get_size(phi));
+  if (m_max < 2) {
+    ERROR("Must use m_max>=2, not m_max=" << m_max);
+  }
+
+  // `DataVectors` for working. `pbar_factor` is guaranteed to be at least size
+  // 3 as demanded by the `YlmSpherePack` constructor
+  T sinmphi(num_points_);
```

Comment:
for my understanding, when `T` is a double this means `sinmphi` is set to `num_points_`. Is that an issue and/or would a brief C++ comment acknowledging this be helpful?

Reply:
Yeah, a comment would be nice.  I don't think it's an issue though, since sinmphi (and cosmphi) are reset below.

Reply:
I agree and added a comment for clarity.